### ISSUE TITLE
Mapgen problems

### ISSUE THREAD

**Issue by ObaniGemini** [MinetestForFun/server-minetestforfun-magichet#27]
- Normal trees are very difficult to find
- Too many empty biomes, it's very difficult to get ressources and very long to travel.
- Sometimes oceans aren't generated properly, and are brutally cut (making some 8/10 blocks falls in a desert land)

**Comment by Ombridride** [comment]
1) The biome is like this in some areas
2) The biome is like this in some areas
3) A mapgen bug, need to check/fix it

**Comment by ObaniGemini** [comment]
I know the biome are like this in some areas. But that's not a reason to not tweak it. Because of some other features or problems, it's very difficult to play magichet. It's harder than mff hardcore server !

**Comment by Ombridride** [comment]
Assignee @4aiman because it's an Upstream issue and he is the best dev to fix it because he knows the mods better than us

**Comment by 4aiman** [comment]
As I've said I'm not happy with the mapgen.

But this will have to wait for the "Summon" feature merge.
By that time the mapgen is to be tweaked anyway, and I'm not fancy doing that twice.

But:
1. That's an issue. Only healing and frost trees need tools to cut those. However, adding more apple-trees is trivial.
2. Not an issue either. Some variety can be added, but not now. There's plenty of food everywhere (onion) and of building materials which are easy-to-get (dirt). 
3.That IS a feature. Desert biomes (there're 2 or 3 of those) have level of water a bit lower than the ocean level (-10 nodes). I actually like how those look like. I can add a fix for borders (there will be a wall of some material that defines the edge of a desert biome), but not now.

**Comment by Ombridride** [comment]
I think it's what @ObaniGemini is talking about : 
![strange_mapgen_liquid](http://i.imgur.com/ablDPls.jpg)

**Comment by ObaniGemini** [comment]
@Ombridride : There are many other places with that problem. (I found two others by travelling a little bit)

@4aiman : is dirt really a building material ?...

**Comment by 4aiman** [comment]
The screenie is entirely normal. (see fiery biome on the right side of a SS above)
The cavegen pierced through water is not (water needs to be made non-ground_content).

But the water won't be updated w/o something triggering that update. So that's too is more than normal. Add some ```minetest.after()``` to on_generated to update water ;)


@ObaniGemini why not? Always make a dirt hut prior to anything else. Hunger won't kill you if you won't move (the food_exhaustion delta is very small for idle).

**Comment by Ombridride** [comment]
@4aiman 
Yes, we need to have the latest fix you done for your subgame, if we don't have them we can't give you the good bug reports... it's a logic...

**Comment by 4aiman** [comment]
On the contrary.
Dev branch contains many experimental things. Bugs found in master help to make dev more stable.
I'll provide you with anything that can be backported nevertheless, but there are so few fixes which can be applied w/o upgrading master to dev...

Meanwhile, you're (e.g. MFF team) is changing the game anyway and sometimes the fix will be your-version-specific. I'm busy enough with MGC dev to support mff version alongside.

What I need is to know who of you is the most experienced with MT Lua API so that things like:

>Add some ```minetest.after()``` to on_generated to update water ;)

would be addressed to a person who can use them and "fix" something.
